Source/WebCore/dom/Document.cpp

```diff
@@ -2126,7 +2126,11 @@ void Document::resolveStyle(ResolveStyleType type)
 
         m_inStyleRecalc = true;
 
-        if (m_needsFullStyleRebuild)
+        bool needsFullStyleRebuild = m_needsFullStyleRebuild;
+#if ENABLE(FULLSCREEN_API)
+        needsFullStyleRebuild = needsFullStyleRebuild || m_fullscreenManager->isResizing();
```

Comment:
Seems odd that stye resolves needs to know anything about fullscreen. Why doesn't window resize have the same issue? I think we should debug further.

Reply:
I can also fix this by calling `scheduleFullStyleRebuild` in a few places but it is a slightly larger change. Do you think that would be reasonable? I can try that instead.

Reply:
Fwiw, full style rebuilds were originally there because we had the UA styles matching the `:-webkit-full-screen-ancestor` pseudo class requiring _all_ the ancestors of the fullscreen element to have their styles recomputed.

It is in theory no longer required with the new fullscreen API (for which I did remove full style rebuilds in some places but not all). The remaining ones avoid dirty renderer assertions when pushing to top layer (which we should arguably fix and then remove the full style rebuilds).

Adding more full style rebuilds isn't really the answer here. Fine grained PseudoClassChangeInvalidation should do its job here.